File: nhglib-core/src/io/github/movementspeed/nhglib/input/enums/TouchInputType.java
package io.github.movementspeed.nhglib.input.enums;

public enum TouchInputType {
    TAP,
    DRAG,
    PINCH,
    ZOOM;

    public static TouchInputType fromString(String value) {
        TouchInputType res = TAP;

        switch (value.toLowerCase()) {
            case "tap":
                break;

            case "drag":
                res = DRAG;
                break;

            case "pinch":
                res = PINCH;
                break;

            case "zoom":
                res = ZOOM;
                break;
        }

        return res;
    }
}

File: nhglib-core/src/io/github/movementspeed/nhglib/input/enums/TouchInputType.kt
package io.github.movementspeed.nhglib.input.enums

enum class TouchInputType {
    TAP,
    DRAG,
    PINCH,
    ZOOM;

    companion object {
        fun fromString(value: String): TouchInputType {
            var res = TAP

            when (value.toLowerCase()) {
                "drag" -> res = DRAG
                "pinch" -> res = PINCH
                "zoom" -> res = ZOOM
            }

            return res
        }
    }
}
